File: backend/app/models/brand.py
from datetime import datetime
from sqlalchemy import Boolean, DateTime, Float, ForeignKey, JSON, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.db.base import Base, TimestampMixin


class BrandWatch(Base, TimestampMixin):
    __tablename__ = "brand_watches"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    original_url: Mapped[str] = mapped_column(String(500), nullable=False)
    keywords: Mapped[list | None] = mapped_column(JSON, nullable=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    similarity_threshold: Mapped[float] = mapped_column(Float, default=70.0, nullable=False)
    scan_schedule: Mapped[str | None] = mapped_column(String(100), nullable=True)
    last_scan: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    user: Mapped["User"] = relationship("User")  # type: ignore[name-defined]
    alerts: Mapped[list["BrandAlert"]] = relationship(
        "BrandAlert", back_populates="brand_watch", cascade="all, delete-orphan"
    )


class BrandAlert(Base):
    __tablename__ = "brand_alerts"

    id: Mapped[int] = mapped_column(primary_key=True)
    brand_watch_id: Mapped[int] = mapped_column(
        ForeignKey("brand_watches.id", ondelete="CASCADE"), nullable=False
    )
    found_domain: Mapped[str] = mapped_column(String(500), nullable=False)
    similarity_score: Mapped[float] = mapped_column(Float, nullable=False)
    detection_sources: Mapped[list | None] = mapped_column(JSON, nullable=True)
    screenshot_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    status: Mapped[str] = mapped_column(String(50), default="new", nullable=False)  # new/reviewed/dismissed
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=__import__("sqlalchemy").func.now(),
        nullable=False,
    )

    brand_watch: Mapped["BrandWatch"] = relationship("BrandWatch", back_populates="alerts")

File: frontend/src/api/brand.ts
import apiClient from './client'

export interface BrandWatch {
  id: number
  name: string
  original_url: string
  keywords: string[] | null
  description: string | null
  similarity_threshold: number
  scan_schedule: string | null
  last_scan: string | null
  is_active: boolean
  alert_count: number
  new_alert_count: number
  created_at: string
}

export interface BrandAlert {
  id: number
  brand_watch_id: number
  found_domain: string
  similarity_score: number
  detection_sources: string[] | null
  screenshot_url: string | null
  status: 'new' | 'reviewed' | 'dismissed'
  notes: string | null
  created_at: string
}

export interface BrandReport {
  brand_watch: BrandWatch
  alerts: BrandAlert[]
  generated_at: string
  summary: {
    total_alerts: number
    new: number
    reviewed: number
    dismissed: number
    avg_similarity: number
  }
}

export const brandApi = {
  listWatches: () => apiClient.get<BrandWatch[]>('/brand/watches'),
  getWatch: (id: number) => apiClient.get<BrandWatch>(`/brand/watches/${id}`),
  createWatch: (data: {
    name: string
    original_url: string
    keywords?: string[]
    description?: string
    similarity_threshold?: number
    scan_schedule?: string
  }) => apiClient.post<BrandWatch>('/brand/watches', data),
  updateWatch: (id: number, data: Partial<BrandWatch>) =>
    apiClient.put<BrandWatch>(`/brand/watches/${id}`, data),
  deleteWatch: (id: number) => apiClient.delete(`/brand/watches/${id}`),
  triggerScan: (id: number) => apiClient.post(`/brand/watches/${id}/scan`),
  listAlerts: (watchId: number) => apiClient.get<BrandAlert[]>(`/brand/watches/${watchId}/alerts`),
  updateAlertStatus: (alertId: number, status: string, notes?: string) =>
    apiClient.put<BrandAlert>(`/brand/alerts/${alertId}/status`, { status, notes }),
  getReport: (watchId: number) => apiClient.get<BrandReport>(`/brand/report/${watchId}`),
}
